client/hack/update-codegen.sh

```diff
@@ -20,6 +20,11 @@ set -o pipefail
 
 CLIENT_ROOT=$(unset CDPATH && cd "$(dirname "${BASH_SOURCE[0]}")"/.. && pwd)
 
+# because kubernetes/code-generator is used as a bash script instead of a go binary, it has to be
+# downloaded. kubernetes/code-generator project recommends vendor dir for this
+# this unfortunately requires downloading unnecessary other deps, but client has few
+go mod vendor
+
```

Comment:
This is probably the most unexpected part of what changed in this PR. The code-generator repo needs downloaded somewhere to run codegen, and vendoring is the most straightforward way to do that I could find. I added this vendor dir to gitignore, and the end of this script removes the dir when finished.